toml/src/main/kotlin/org/rust/toml/resolve/CargoTomlFileReferenceProvider.kt

```diff
@@ -67,4 +78,21 @@ private class CargoTomlFileReferenceSet(
             }
         }
     }
+
+    override fun createFileReference(range: TextRange?, index: Int, text: String?): FileReference? {
+        val reference = super.createFileReference(range, index, text) ?: return null
+
+        return if (isGlobPathFragment(text)) {
+            GlobReference(reference)
```

Comment:
Why do we need `GlobReference` at all? I think we can return `null` if `isGlobPathFragment(text)`

Reply:
I did that originally, but if you return null, there will be no way to find out that a glob has occurred "in the middle". Therefore we could not skip path fragments after a glob. Maybe it could be done with some inner state inside the reference set, but this seemed cleaner.

Reply:
Oh, I see
But I'm still a bit worried about the current implementation because if `createFileReference` doesn't return null, `FileReferenceSet` saves the reference into inner array and it can be used in other parts.
So I thought about saving index of the first glob pattern into local field and checking it in `createFileReference`.
To restore the default state, we can override `reparse`
Something like
```kotlin
private var firstGlobPatternIndex: Int? = null

override fun reparse() {
    try {
        super.reparse()
    } finally {
        firstGlobPatternIndex = null
    }
}
```

Reply:
```kotlin
override fun reparse() {
    firstGlobPatternIndex = null
    super.reparse()
}
```
This makes more sense to me - the state should be reset before every parse, right? Or maybe I misunderstood.

Reply:
Oh, it doesn't matter, they do the same things